test/jdk/java/foreign/TestLayoutPaths.java

```diff
@@ -489,5 +493,53 @@ public void testOffsetHandle(MemoryLayout layout, PathElement[] pathElements, lo
         return testCases.toArray(Object[][]::new);
     }
 
+    @Test(dataProvider = "testLayouts")
+    public void testSliceHandle(MemoryLayout layout, PathElement[] pathElements, long[] indexes,
+                                long expectedBitOffset) throws Throwable {
+        if (expectedBitOffset % 8 != 0)
+            throw new SkipException("Offset not a multiple of 8");
+
+        MemoryLayout selected = layout.select(pathElements);
+        MethodHandle sliceHandle = layout.sliceHandle(pathElements);
+        sliceHandle = sliceHandle.asSpreader(long[].class, indexes.length);
+
+        try (ResourceScope scope = ResourceScope.ofConfined()) {
+            MemorySegment segment = MemorySegment.allocateNative(layout, scope);
+            MemorySegment slice = (MemorySegment) sliceHandle.invokeExact(segment, indexes);
+            assertEquals(slice.address().segmentOffset(segment), expectedBitOffset / 8);
+            assertEquals(slice.byteSize(), selected.byteSize());
+        }
+    }
+
+    @Test(expectedExceptions = UnsupportedOperationException.class)
+    public void testSliceHandleUOEInvalidSize() {
+        MemoryLayout layout = MemoryLayout.ofStruct(
+            MemoryLayout.ofValueBits(32, ByteOrder.nativeOrder()).withName("x"),
+            MemoryLayout.ofValueBits(31, ByteOrder.nativeOrder()).withName("y") // not a multiple of 8
+        );
+
+        layout.sliceHandle(groupElement("y")); // should throw
+    }
+
+    @Test(expectedExceptions = UnsupportedOperationException.class)
+    public void testSliceHandleUOEInvalidOffset() throws Throwable {
+        MemoryLayout layout = MemoryLayout.ofStruct(
+            MemoryLayout.ofPaddingBits(5),
+            MemoryLayout.ofValueBits(32, ByteOrder.nativeOrder()).withName("y") // offset not a multiple of 8
+        );
+
+        MethodHandle sliceHandle;
+        try {
+            sliceHandle = layout.sliceHandle(groupElement("y")); // should work
```

Comment:
Question: why is the failure dynamic? We know exactly, at construction time that the offset is gonna be odd?

Reply:
That is true in this case, but not in general where the offset could depend on some dynamic indexes.

However, I could change it to eagerly do the offset check in case we don't have any free dimensions.

Reply:
ugh - you mean accessing something like 3bit value array in the 8th element? Yeah...

Reply:
Yes, exactly. And e.g. if we always pick 0 as offset but the stride is not a multiple of 8 we would not catch that either.

I've added an eager check if there are no strides, and added a test case as well.